clients/src/main/java/org/apache/kafka/clients/consumer/internals/TopicMetadataRequestManager.java

```diff
@@ -96,7 +96,7 @@ public NetworkClientDelegate.PollResult poll(final long currentTimeMs) {
      * @param topic to be requested. If empty, return the metadata for all topics.
      * @return the future of the metadata request.
      */
-    public CompletableFuture<Map<String, List<PartitionInfo>>> requestTopicMetadata(final Optional<String> topic) {
+    public CompletableFuture<Map<Topic, List<PartitionInfo>>> requestTopicMetadata(final Optional<String> topic) {
```

Comment:
Do we still need the changes in this class? It seems that we don't use them anymore. I have the same question for the new `Topic` class.

Reply:
You're right, not needed anymore (all metadata interaction is now based on the centralized metadata cache). All removed.